lib/matplotlib/__init__.py

```diff
@@ -119,10 +119,7 @@
 import contextlib
 import distutils.version
 import functools
-import io
-import importlib
 import inspect
-from inspect import Parameter
```

Comment:
Leave this import?  (I tend to import classes (and nothing else) by unqualified name, as a rule.)

Reply:
Do we have a policy on this? It seems a bit imbalanced that we do

~~~
import inspect
from inspect import Parameter

inspect.signature()
Parameter
~~~

Generally, I would recommend that we do not import individual classes if they are only rarely used. Importing classes and functions pollutes the namespace more. Also, when it's just rarely used, I usually wouldn't know what `Parameter` is, whereas `inspect.Parameter` is quite clear.

Reply:
https://www.python.org/dev/peps/pep-0008/#imports

> When importing a class from a class-containing module, it's usually okay to spell this:

    from myclass import MyClass
    from foo.bar.yourclass import YourClass

> If this spelling causes local name clashes, then spell them explicitly:

    import myclass
    import foo.bar.yourclass

> and use "myclass.MyClass" and "foo.bar.yourclass.YourClass".